File: practicas/practica_2/Deepspace_Ruby_P2/lib/GameState.rb
#encoding:utf-8

# File: GameState.rb
# Author: Profesor

module Deepspace

# Brief: Representa el estado del juego    
module GameState
  
    CANNOTPLAY = :cannotplay
    INIT = :init
    BEFORECOMBAT = :beforecombat
    AFTERCOMBAT = :aftercombat
    

#     CANNOTPLAY:     the Gameuniverse has been created but not initialized
#     INIT:   the GameUniverse has been initialized and the current player has not combated. Players can mount weapons and shielboosters, or combat
#     BEFORECOMBAT:   players can only combat
#     AFTERCOMBAT:    players can mount/discard weapons,shielboosters and hangars 

end # Module GameState

end # Module Deepspace

File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/GameState.java
/**
 * @file GameState.java
 * @author Profesor
 */
package deepspace;

/**
 * @brief Representa el estado del juego
 */
public enum GameState {
    CANNOTPLAY, INIT,BEFORECOMBAT,AFTERCOMBAT
    /*  
    CANNOTPLAY:     the Gameuniverse has benn created but not initialized
    INIT:   the GameUniverse has been initialized and the current player has not combated. 
            Players can mount weapons and shielboosters, or combat
    BEFORECOMBAT:   players can only combat
    AFTERCOMBAT:    players can mount/discard weapons,shielboosters and hangars 
    */
}

//    INIT:   the GameUniverse has been initialized and the current player has not combated. 
//            Players can mount/discard weapons,shielboosters and hangars or combat